Simplenote/Classes/SPNoteEditorViewController+Extensions.swift

```diff
@@ -153,13 +153,76 @@ extension SPNoteEditorViewController {
     /// - Parameter button: The button that triggered the action
     @objc
     func handleNoteOptions(_ button: UIButton) {
-        let noteView = NoteOptionsViewController(style: .grouped)
+
+        save()
+        endEditing(button)
+        tagView.endEditing(true)
+
+        SPTracker.trackEditorActivitiesAccessed()
+
+        let noteView = NoteOptionsViewController(with: currentNote)
+        noteView.delegate = self
         let noteNavigation = SPNavigationController(rootViewController: noteView)
         noteNavigation.displaysBlurEffect = true
         noteNavigation.modalPresentationStyle = .popover
         noteNavigation.popoverPresentationController?.sourceRect = button.bounds
         noteNavigation.popoverPresentationController?.sourceView = button
         noteNavigation.popoverPresentationController?.backgroundColor = .simplenoteNavigationBarModalBackgroundColor
+        noteNavigation.popoverPresentationController?.delegate = self
         present(noteNavigation, animated: true, completion: nil)
```

Comment:
We could, perhaps, encapsulate a bit, and split this method in two, WDYT?

Reply:
Yeah it's getting a bit busy. Perhaps I could leave the save method call and tracking here and then move the rest into a `presentNoteOptions` method?

Reply:
Looks good! Yeah it was a bit too overloaded (single API doing lots of things). Looking good!